scripts/docker-smoke-test.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# docker-smoke-test.sh
+# Start the containerized api and run some end-to-end tests against it
+# There will be some overlap with just running a TestClient in the unit tests
+# Is there a good way to reuse code here?
+# Also note this can evolve into a generalized pipeline smoke test
+
+# shellcheck disable=SC2317  # Shellcheck complains that trap functions are unreachable...
```

Comment:
Oh, good call out. Given that it's only a couple of tests right now, feels okay-ish, but would be nice to have a pattern here if it makes sense and we're going to be doing this more often.